### PR TITLE
[12.0][IMP] stock_request_analytic + stock_request_purchase: Add Analytic Tags and propagate to PO

### PR DESCRIPTION
**Pull request by joao-p-marques**
Needs https://github.com/OCA/account-analytic/pull/366 to add `analytic_tag_ids` field to `stock.move`.

@Tecnativa
TT28488

ping @pedrobaeza @chienandalu

### CODE REVIEW

**Review by rousseldenis — APPROVED** [review]
LGTM

**Comment by rousseldenis** [comment]
@joao-p-marques Could you check travis ?

**Comment by joao-p-marques** [comment]
Thanks @rousseldenis 
Should be fixed now :+1:

**Comment by rousseldenis** [comment]
> Thanks @rousseldenis
> Should be fixed now 👍

Nearly. One test still failing

**Comment by joao-p-marques** [comment]
> Nearly. One test still failing

Oh sorry. Yes, it is due to the fact that we need https://github.com/OCA/account-analytic/pull/366 to add the `analytic_tag_ids` field to `stock.move`, as mentioned above.

Once that is merged, I will relaunch the build here, but the tests are successful locally with the mentioned changes.

File: stock_request_analytic/models/stock_request.py [hunk 1]
```diff
@@ -11,6 +11,9 @@ class StockRequest(models.Model):
     analytic_account_id = fields.Many2one(
         'account.analytic.account', string='Analytic Account')
```
Comment (pedrobaeza) [inline stock_request_analytic/models/stock_request.py]: No need of this empty line

File: stock_request_analytic/models/stock_rule.py [hunk 2]
```diff
@@ -15,5 +15,10 @@ def _get_stock_move_values(self, product_id, product_qty, product_uom,
         if values.get('stock_request_id'):
             analytic_account_id = self.env['stock.request'].browse(
                 values['stock_request_id']).analytic_account_id.id
-            res.update(analytic_account_id=analytic_account_id)
+            analytic_tag_ids = self.env['stock.request'].browse(
```
Comment (pedrobaeza) [inline stock_request_analytic/models/stock_rule.py]: Don't browse twice, but store it in an intermediate variable and browse once.

File: stock_request_analytic/tests/test_stock_request_analytic.py [hunk 3]
```diff
@@ -42,8 +42,9 @@ def setUp(self):
         })
         self.pizza.route_ids = [(6, 0, self.demand_route.ids)]
 
-    def prepare_order_request_analytic(self, aa, company):
+    def prepare_order_request_analytic(self, aa, company, ats=None):
```
Comment (pedrobaeza) [inline stock_request_analytic/tests/test_stock_request_analytic.py]: Significant argument names

```suggestion
    def prepare_order_request_analytic(self, aa, company, analytic_tags=None):
```

File: stock_request_purchase/models/stock_rule.py [hunk 4]
```diff
@@ -12,7 +12,14 @@ def _prepare_purchase_order_line(self, product_id, product_qty,
...
             product_id, product_qty, product_uom, values, po, supplier)
         if 'stock_request_id' in values:
-            vals['stock_request_ids'] = [(4, values['stock_request_id'])]
+            stock_request = self.env["stock.request"].browse(values['stock_request_id'])
+            vals['stock_request_ids'] = [(4, stock_request.id)]
+            if stock_request.analytic_account_id:
+                vals['account_analytic_id'] = stock_request.analytic_account_id.id
+            if stock_request.analytic_tag_ids:
+                vals['analytic_tag_ids'] = [
+                    (4, ati) for ati in stock_request.analytic_tag_ids.ids
```
Comment (pedrobaeza) [inline stock_request_purchase/models/stock_rule.py]: More significant variable names and code readability:

```suggestion
                    (4, tag.id) for tag in stock_request.analytic_tag_ids
```

File: stock_request_analytic/tests/test_stock_request_analytic.py [hunk 5]
```diff
@@ -42,8 +42,9 @@ def setUp(self):
...
         expected_date = fields.Datetime.now()
+        ats_ids = ats.ids if ats else []
```
Comment (pedrobaeza) [inline stock_request_analytic/tests/test_stock_request_analytic.py]:
```suggestion
        if analytic_tags is None:
            analytic_tags = env["account.analytic.tag"]
```

File: stock_request_analytic/tests/test_stock_request_analytic.py [hunk 6]
```diff
@@ -54,6 +55,7 @@ def prepare_order_request_analytic(self, aa, company):
                 'product_uom_id': self.pizza.uom_id.id,
                 'product_uom_qty': 5.0,
                 'analytic_account_id': aa.id,
+                'analytic_tag_ids': [(4, ati) for ati in ats_ids],
```
Comment (pedrobaeza) [inline stock_request_analytic/tests/test_stock_request_analytic.py]: More significant variable names and code readability

```suggestion
                'analytic_tag_ids': [(4, tag.id) for tag in analytic_tags],
```

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

**Comment by joao-p-marques** [comment]
@pedrobaeza changes made. Thanks :+1:

**Comment by pedrobaeza** [comment]
@joao-p-marques I think you haven't pushed the updated branch

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**